src/main/java/com/kaleido/fetch/service/ActivityService.java

```diff
@@ -125,18 +134,39 @@ public Activity getActivity(Long id) {
         return restTemplate.exchange(plateMapURI, HttpMethod.PUT, entity, PlateMap.class);
     }
     
-    public PlateMap getActivitiesPlatemap(String activityName) {
+    public ResponseEntity<PlateMap[]> getActivitiesPlatemap(PlateMap plateMap) {
     	
+    	log.info("PlateMap data is ", plateMap);
+        String plateMapURI = cabinetURI + "plate-maps/details";
+    	
+    	RestTemplate restTemplate = new RestTemplate();
+    	HttpHeaders headers = new HttpHeaders();
+    	headers.setContentType(MediaType.APPLICATION_JSON);
+    	HttpEntity<PlateMap> entity = new HttpEntity<PlateMap>(plateMap,headers);
+        return restTemplate.exchange(plateMapURI, HttpMethod.POST, entity, PlateMap[].class);
+    }
+    
+    public ResponseEntity<PlateMap[]> getActivitiesList(String activityName) {
     	log.info("Activity name is ", activityName);
-        String plateMapURI = cabinetURI + "plate-maps";
+        String plateMapURI = cabinetURI + "plate-maps/details";
     	
     	RestTemplate restTemplate = new RestTemplate();
-        return restTemplate.getForObject(plateMapURI, PlateMap.class);
+    	HttpHeaders headers = new HttpHeaders();
+    	String username = "admin";
+    	String password = "admin";
+    	headers.setBasicAuth(username, password);
```

Comment:
Since it seems like we have a bunch of code copy/paste, should we abstract this out to a cabinet client? For the first iteration, I don't think we need a separate project/dependency, however we should probably make it it's own class.

Reply:
Yes, that would be better